File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void level(TreeNode* node, int& max){
        if(node==NULL){
            return;
        }

        queue<TreeNode*> q;
        q.push(node);
        int lvl = 0;
        int maxsum = INT_MIN;
        while(!q.empty()){
            int size = q.size();
            int sum = 0;
            lvl++;
            for(int i=0;i<size;i++){
                TreeNode* curr = q.front();
                q.pop();
                sum += curr->val;
                if(curr->left!=NULL){
                    q.push(curr->left);
                }
                if(curr->right!=NULL){
                    q.push(curr->right);
                }
            }
            if(sum>maxsum){
                max = lvl;
                maxsum = sum;
                cout<<sum<<endl;
            }
        }
    }
    int maxLevelSum(TreeNode* root) {
        int max = 0;
        level(root,max);
        return max;
        
    }
};

File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int bfs(TreeNode root){
        if(root==null){
            return 0;
        }
        int maxm = Integer.MIN_VALUE;
        int ans = 0;
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        int lvl = 1;
        while(!q.isEmpty()){
            int l = q.size();
            int sum = 0;
            for(int i=0;i<l;i++){
                TreeNode curr = q.poll();
                sum+=curr.val;
                if(curr.left!=null){
                    q.add(curr.left);
                }
                if(curr.right!=null){
                    q.add(curr.right);
                }
            }
            if(sum>maxm){
                maxm = sum;
                ans = lvl;
            }
            lvl++;
        }
        return ans;
    }
    public int maxLevelSum(TreeNode root) {
        return bfs(root);
    }
}